src/Microsoft.DotNet.Interactive.VSCode/VSCodeInteractiveHost.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Interactive.Events;
+
+namespace Microsoft.DotNet.Interactive.VSCode
+{
+    public class VSCodeInteractiveHost : IInteractiveHost
+    {
+        /// <summary>
+        /// Gets input from the user.
+        /// </summary>
+        /// <param name="prompt">The prompt to show.</param>
+        /// <param name="isPassword">Whether the input should be treated as a password.</param>
+        /// <returns>The value.</returns>
+        public Task<string> GetInputAsync(string prompt = "", bool isPassword = false, CancellationToken cancellationToken = default)
+        {
+            var command = new GetInput(prompt, isPassword, targetKernelName: "vscode");
+            var completionSource = new TaskCompletionSource<string>();
+            var token = command.GetToken();
+            Kernel.Root.KernelEvents.Where(e => e.Command.GetToken() == token).Subscribe(e =>
```

Comment:
We can typically reference the kernel directly (e.g. pass it into the constructor) rather than relying on the static method, which in turn relies on `AsyncLocal`.

Reply:
Added a commit where we directly pass in and use the kenel.

Reply:
Done.